### PR TITLE
feat: ripioPortalWallet support

### PR DESCRIPTION
**Pull request by nicolaswww**
Add support for https://www.ripio.com/portal/

Ripio Portal is detected as MetaMask: when it's installed (only on some dApps), both isMetaMask and isPortal are set to true. In most cases isPortal will be true and isMetaMask false.

### CODE REVIEW

File: packages/rainbowkit/src/wallets/walletConnectors/metaMaskWallet/metaMaskWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/metaMaskWallet/metaMaskWallet.ts:36]: Revert these changes, or provide additional documentation about the conflict. As long as the Ripio connector is available in the wallet list, this injected flag conflict should be automatically resolved.

File: packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts:6]: Please adjust `BraveWalletOptions` type

File: packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts:15]: Let's use `ripio` or `ripioPortal` id instead

File: packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts:28]: Expecting WalletConnect connector support for Ripio mobile here as well. Reference MetaMask connector for a multi-platform connector implementation

File: site/components/Provider/Provider.tsx
Comment (DanielSinclair) [inline site/components/Provider/Provider.tsx:41]: Please revert these changes to the docs site. Instead, add to `packages/example` example site. This connector will be testable at a Vercel deployment once approved; this is our typical review process.

File: packages/rainbowkit/src/wallets/walletConnectors/index.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/index.ts:23]: Please use name `ripioWallet` or `ripioPortalWallet`. Ripio namespace is preferred if this connector will support the mobile wallet as well

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

File: packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts
Comment (nicolaswww) [reply packages/rainbowkit/src/wallets/walletConnectors/portalWallet/portalWallet.ts]: Currently our Ripio Portal wallet is only an extension (it is independent of the Ripio mobile app). I removed the QR code and only left the URL of the extension to avoid confusion.

**Comment by nicolaswww** [comment]
Thanks for the requests @DanielSinclair. I already applied all the requested changes and added Ripio Portal to `packages/example`.

File: packages/rainbowkit/src/wallets/walletConnectors/ripioPortalWallet/ripioPortalWallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/ripioPortalWallet/ripioPortalWallet.ts:23]: This should be the Chrome Web Store URL

File: packages/rainbowkit/src/wallets/walletConnectors/ripioPortalWallet/ripioPortalWallet.svg
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/ripioPortalWallet/ripioPortalWallet.svg:1]: This should preferably be larger than 32x32

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Attempted connection with the Ripio Portal extension on the example dApp [here](https://rainbowkit-example-41mm1ivjz-rainbowdotme.vercel.app/), but was unable to login. Created account, attempted authentication in the extension, am redirected to login page and redirected again to this page, where the extension launches again. Am unable to break this cycle.
Screenshot 2023-02-07 at 2 06 00 AM

**Comment by DanielSinclair** [comment]
@nicolaswww Unable to successfully connect a wallet. Attempt to auth in the new window popup, am redirect to Ripio, a few new tabs open, and then popup reset and I attempt to auth again, repeating the issues. Will close this for now, but if the stability improves and you think there is a viable web wallet approach, will revisit integration discussion.

**Comment by nicolaswww** [comment]
> @nicolaswww Unable to successfully connect a wallet. Attempted to auth in the new window popup, am redirected to Ripio, a few new tabs open, and then the popup resets and I attempt to auth again, repeating the issues. Will close this for now, but if the stability improves, we can revisit the integration discussion.

Sorry for the delay @DanielSinclair, on Wednesday we will release a new version of our extension that fixes that issue, when that happens I'll open the request again. Thank you!

(if you want to try it before that, just open [kyc.ripio.com](https://kyc.ripio.com) instead of [mkyc.ripio.com](https://mkyc.ripio.com))

**Comment by nicolaswww** [comment]
Hi @DanielSinclair, we already solved the problems with the extension. Could you reopen the pull request and try again? thanks!

**Comment by DanielSinclair** [comment]
@nicolaswww Seeing a CI error. Please run `pnpm run lint:fix`

**Comment by DanielSinclair** [comment]
@nicolaswww Unable to QA the extension from the United States. Do you happen to have a demo account available? Feel free to email me privately at `daniel@rainbow.me` to coordinate this QA
Screenshot 2023-02-25 at 4 13 30 PM

**Comment by nicolaswww** [comment]
Thanks @DanielSinclair, I have already made the requested lint change and requested an account to be used in the US, I will share the access as soon as it is sent to me.

**Comment by nicolaswww** [comment]
Hi @DanielSinclair ! I sent you an email this week to coordinate the QA, we already have a test account. Did you get to see it? Thank you!

**Comment by nicolaswww** [comment]
Hi @DanielSinclair ! We have adjusted the MR, could you review it and tell us if we should send something else? Thank you!

**Comment by nicolaswww** [comment]
Hi @DanielSinclair , Is there something we should do to continue with this PR? thank you!

**Comment by DanielSinclair** [comment]
@nicolaswww Please enabled changes from maintainers on this PR

**Pull request closed without merging**
